QModManager/Patching/IManifestValidator.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace QModManager.Patching
+{
+    internal interface IManifestValidator
```

Comment:
What's the point of dependency injection for `internal` classes?

Reply:
Unit testing?

Reply:
Yes, and also making it clear how the class is supposed to be used.

There are internal methods that are accessed directly by unit tests,  
but for normal use, only the interface methods should be getting accessed by external actors.